### PR TITLE
Document reprovision param for Update Workflow API

### LINKED ISSUE #8166: [DOC] Add documentation for Flow Framework Reprovision parameter
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.

A reprovision parameter was added to Flow Framework update template (PUT) API:
 - https://github.com/opensearch-project/flow-framework/pull/804

**Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.

New in 2.17.0
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

Linked PR has pretty through description of the feature.

Sorry this was missed; I'll get a PR up soon.

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Adds documentation for the Update Workflow API `reprovision` parameter

### Issues Resolved

Resolves #8166

### Version

2.17.0

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbwiddis** [comment]
@amitgalitz @ohltyler Can you please provide a tech review for this since @joshpalis is unavailable and you either reviewed the PR or have used the feature from the front end?
 - https://github.com/opensearch-project/flow-framework/pull/804

File: _automating-configurations/api/create-workflow.md
Comment (dbwiddis) [inline _automating-configurations/api/create-workflow.md:75]: I expect this section to create a merge conflict when #8171 is merged. This section should go above that new content as it's directly related to updating.  I'll resolve conflicts when the time comes.

File: _automating-configurations/api/create-workflow.md
Comment (amitgalitz) [inline _automating-configurations/api/create-workflow.md:86]: The {: .note} is not producing anything specific in markdown, can we remove?

File: _automating-configurations/api/create-workflow.md
Comment (amitgalitz) [inline _automating-configurations/api/create-workflow.md:83]: can be removed I assume

**Review by ohltyler — COMMENTED** [review]
Nothing to add from @amitgalitz comments. Will approve once those are resolved

File: _automating-configurations/api/create-workflow.md
Comment (dbwiddis) [reply _automating-configurations/api/create-workflow.md:86]: I think that renders on the actual documentation page, see the similar one above which renders like this:

Screenshot 2024-09-04 at 4 51 44 PM

File: _automating-configurations/api/create-workflow.md
Comment (dbwiddis) [reply _automating-configurations/api/create-workflow.md:83]: I'll leave that to doc reviewers.  I see the `<placeholder>` stuff used in other copy/paste.

**Comment by dbwiddis** [comment]
> Nothing to add from @amitgalitz comments. Will approve once those are resolved

I don't think there's anything to do on those. :)

File: _automating-configurations/api/create-workflow.md
Comment (amitgalitz) [reply _automating-configurations/api/create-workflow.md:86]: I see, I am okay with it then

File: _automating-configurations/api/create-workflow.md
Comment (amitgalitz) [reply _automating-configurations/api/create-workflow.md:83]: Yeah I see that now in other locations

**Review by ohltyler — APPROVED** [review]

File: _automating-configurations/api/create-workflow.md
Comment (dbwiddis) [reply _automating-configurations/api/create-workflow.md:86]: I anticipate we'll make this particular paragraph more detailed and link to a whole separate page (like the use case templates) for the remainder of the feature, so just trying to add the minimum this time.

File: _automating-configurations/api/create-workflow.md
Comment (kolchfa-aws) [reply _automating-configurations/api/create-workflow.md:83]: We typically want to spell out everything in the documentation in order to avoid confusion. This request clearly shows the `reprovision` query parameter and specifies that it needs to be a _complete_ template, so we should keep this here. The template is very big so it's fine to use a descriptive placeholder.

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @dbwiddis! Just a couple of suggestions, otherwise LGTM.

**Comment by kolchfa-aws** [comment]
Applying "Done but waiting to merge" label to put this in the 2.17 editorial queue starting 9/11. This PR will be merged before the release.

**Comment by dbwiddis** [comment]
> Applying "Done but waiting to merge" label to put this in the 2.17 editorial queue starting 9/11. This PR will be merged before the release.

Note that when #8171 is merged, it may create a merge conflict with this. I'll keep an eye out and rebase accordingly.

**Comment by kolchfa-aws** [comment]
Noted. This will likely be merged first because it's a 2.17 PR and will be prioritized higher for editorial review.

File: _automating-configurations/api/create-workflow.md
Comment (natebower) [inline _automating-configurations/api/create-workflow.md:84]:
```suggestion
You can add new steps to the workflow but cannot delete them. Only index setting, search pipeline, and ingest pipeline steps can currently be updated.
```

**Review by natebower — COMMENTED** [review]
@dbwiddis @kolchfa-aws Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**
